bindings/go/runtime/registry.go

```diff
@@ -178,45 +152,85 @@ func (r *Scheme) Decode(data io.Reader, into any) error {
 	return nil
 }
 
-func (r *Scheme) Convert(from any, into any) error {
-	// check if typed is a raw, yaml unmarshalling has its own reflection check so we don't need to do this
-	// before the raw assertion.
-	if raw, ok := from.(*Raw); ok {
-		if _, err := r.TypeForPrototype(into); err != nil && !r.allowUnknown {
-			return fmt.Errorf("%T is not a valid registered type and cannot be decoded: %w", into, err)
-		}
-		fromType, err := GetTypeFromAny(from)
-		if err != nil {
-			return fmt.Errorf("could not get type from prototype: %w", err)
+// Convert transforms one Typed object into another. Both 'from' and 'into' must be non-nil pointers.
+//
+// Special Cases:
+//   - Raw → Raw: performs a deep copy of the underlying []byte data.
+//   - Raw → Typed: unmarshals Raw.Data JSON via json.Unmarshal into the Typed object (if Typed.GetType is registered).
+//   - Typed → Raw: marshals the Typed with json.Marshal, applies canonicalization, and stores the result in Raw.Data.
+//     (See Raw.UnmarshalJSON for equivalent behavior)
+//   - Typed → Typed: performs a deep copy using Typed.DeepCopyTyped, with reflection-based assignment.
+//
+// Errors are returned if:
+//   - Either argument is nil.
+//   - A type is not registered in the Scheme (for Raw conversions).
+//   - A reflection-based assignment fails due to type mismatch.
+func (r *Scheme) Convert(from Typed, into Typed) error {
```

Comment:
The scheme contains a mutex. Don't we need to use that here? 🤔

Reply:
Its used inside the call of IsRegistered which should be fine. WDYT?

Reply:
Hmmm. But there is no defence if someone uses this directly. humm humm.

Reply:
what do you mean with uses it directly? where would we need a lock?